### PR TITLE
Remove dependency to cucumber-wire

### PR DESCRIPTION
**Pull request by aurelien-reeves**
# Description

`cucumber-ruby-wire` has been design to be an external optional dependency. This is not the case yet and it is responsible for adding complexity in the dependency tree for cucumber-ruby.

This PR aims to remove the dependency to `cucumber-ruby-wire`. The wire protocol would still be available but requiring it explicitly.

## Type of change

- Refactoring (improvements to code design or tooling that don't change behaviour)
- Breaking change (will cause existing functionality to not
  work as expected)

## Note to other contributors

Tests may be passing but there are a lot of things to do before merging this PR. We will certainly have to deprecate it first in a new major version before removing it completely in another new one.

For now what remains to be done here is to add a hook for `cucumber-ruby-wire` to be able to install by itself.

# Checklist:

Your PR is ready for review once the following checklist is
complete. You can also add some checks if you want to.

- [x] Tests have been added for any changes to behaviour of the code
- [x] New and existing tests are passing locally and on CI
- [x] `bundle exec rubocop` reports no offenses
- [x] RDoc comments have been updated
- [x] CHANGELOG.md has been updated
- [x] cucumber-ruby-wire is ready and work with this branch
- [x] The wire protocol has been deprectated

### CODE REVIEW

**Comment by aurelien-reeves** [comment]
https://github.com/cucumber/cucumber-ruby-wire/pull/46 is the cucumber-ruby-wire counter-part of this PR.

Everything seems to be ok actually, we would just need a deprecation/release plan for this but technically, it seems to be all good

**Comment by aurelien-reeves** [comment]
We are ready to remove it now 😀

**Comment by aurelien-reeves** [comment]
I am ready to merge that one if no further objection @mattwynne @luke-hill?

File: RELEASING.md [hunk 1]
```diff
@@ -8,16 +8,10 @@ To release `cucumber-ruby`, you'll need:
...
-## Prerequisites
-
-To release `cucumber`, you'll need to be a member of the core team
-
-## cucumber-ruby-core and cucumber-ruby-wire
+## cucumber-ruby-core
 
 If internal libraries such as `cucumber-gherkin` needs to be updated, you'll
-need to update and release `cucumber-ruby-core` first, then `cucumber-ruby-wire`.
+need to update and release `cucumber-ruby-core` before releasing `cucumber-ruby`.
```
Comment (luke-hill) [inline RELEASING.md:14]: Should this say gherkin?

File: UPGRADING.md [hunk 2]
```diff
@@ -1,5 +1,50 @@
 # Upgrading to 8.0.0
 
+# The wire protocol
+
+The built-in wire protocol has been removed.
+
+The wire protocol is still available by explicitely using the `cucumber-wire` gem.
```
Comment (luke-hill) [inline UPGRADING.md:7]: explicitly*

File: UPGRADING.md [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
+  # Gemfile
+
+  # ...
+
+  gem "cucumber"
+  gem "cucumber-wire"
+
+  # ...
+
+  ```
```
Comment (luke-hill) [inline UPGRADING.md:37]: extra triple backticks

**Review by luke-hill — COMMENTED** [review]
Minor typos. Other than that RtM

File: RELEASING.md (hunk 1 above)
Comment (aurelien-reeves) [reply RELEASING.md:14]: The gem is named "cucumber-gherkin": https://rubygems.org/gems/cucumber-gherkin 🤔

File: UPGRADING.md (hunk 3 above)
Comment (aurelien-reeves) [reply UPGRADING.md:37]: Looks good to me

There are the one that close the one for the gemfile, then we open a new block for a shell command 🤔

Comment (aurelien-reeves) [reply UPGRADING.md:37]: image

File: RELEASING.md (hunk 1 above)
Comment (luke-hill) [reply RELEASING.md:14]: yes I mean it says ruby core here mixed with gherkin

Comment (aurelien-reeves) [reply RELEASING.md:14]: Oh, I see.
Actually the idea I wanted to document here is that an update of cucumber-core is required in order to bring a new version of cucumber-gherkin within cucumber-ruby 😅

**Pull request merged**